collector/src/main/java/com/bitmovin/analytics/data/EventData.kt

```diff
@@ -7,14 +7,14 @@ import com.bitmovin.analytics.utils.Util
 class EventData(
     bitmovinAnalyticsConfig: BitmovinAnalyticsConfig,
     val impressionId: String,
-    deviceInfo: DeviceInformation,
     val userId: String
 ) {
-    val userAgent = deviceInfo.userAgent
-    val deviceInformation = DeviceInformationDto(deviceInfo.manufacturer, deviceInfo.model, deviceInfo.isTV)
-    val language: String = deviceInfo.locale //
+    var userAgent: String? = null
+    var deviceInformation: DeviceInformationDto? = null
+    var language: String? = null
```

Comment:
this was non-nullable before?

Reply:
Those parameters have been set in the constructor of the `EventData` before. This was moved to one of the `EventDataDecoratorPipeline`s we introduced in this PR. As a consequence this was changed to be `nullable`.

Reply:
:/ this looks like something that's very prone to mistakes / errors or unintended consequences long term :( 
@rolandgriesser what do you think?

Reply:
Those values were only non-nullable as we retrieved them in the constructor before. I would rather go with one system to *decorate* samples instead of having multiple ways. 
Something we could do: have something a `DeviceInformationDecorator` interface and then after calling the `registerDecorators`, check if this decorator has been provided by the player adapter and otherwise throw an exception. And then pass this decorator in the constructor of `EventData` (and retrieve the properties). But technically we could do this with every available decorator and make them nullable, which would lead to huge constructors and `optional` stuff would be handled really strange.

Reply:
Undid changes to `EventData` in 4a51afec249f3ec25ff41013b01c4ffb321b4ff7 (pass deviceInformation to `EventData`) and renamed `decorators` to `manipulators`